Game controller

j-siderius opened the pull request:
- [x] added timing controller
- [x] added tick-based actions/movement counter, easy to add zombie updates to this
- [x] updated the field drawing to draw all water as one object
- [x] added sky colouring

j-siderius committed:
Merge branch 'Arduino_serial' into GameController

j-siderius committed:
added tick timer implementation (still work in progress)

j-siderius committed:
cleaned the file

j-siderius committed:
added Gamecontroller.py class, manages game ticks and sky color/opacity
implemented new class into main
fixed start_action() in Player.py

j-siderius committed:
updated gitignore
added saving water as static image to Field.py
blitting saved image to the screen every frame
TODO @Frank: do the magic sprite drawing thingy
added sky surface and drawing method

j-siderius committed:
Merge branch 'main' into GameController

j-siderius committed:
implemented LED clock updating
TODO: fix offset of 1

FrankBosman commented on Field.py:
```diff
@@ -103,11 +103,17 @@ def __init__(self, screen, hex_width=25, field_size=None, hex_amount=None):
         self.water_group.draw(screen.get_screen())
+        # save the water sprite as one image for faster loading
+        pygame.image.save(screen.get_screen(), "Data/Sprites/water.bmp")
+        self.water_img = pygame.image.load("Data/Sprites/water.bmp").convert()
```
maybe use surfaces instead, seems to work for me

self.water_surface = pygame.Surface(self.screen.get_size())
self.water_group.draw(self.water_surface)
self.screen.get_screen().blit(self.water_surface, (0,0))

j-siderius replied:
dit werkt bij mij niet: ik krijg een blank/wit water surface

FrankBosman reviewed (commented):
we moeten er nog ff samen naar kijken, want de fps is laag, de overlay ziet er een beetje raar uit en de selectie bolletjes zijn zwart, mis snap ik iets niet

j-siderius committed:
changed day-night colors to only fade to black
added function to screen update loop to call serial (trying to update more frequently)
cleaned up some old code

j-siderius committed:
cleaned up old code in main.py
added potentially quicker serial code to Serial.py
disabled sky draw

j-siderius committed:
enabled sky overlay

j-siderius committed:
added fps counter

j-siderius committed:
Merge remote-tracking branch 'origin/GameController' into GameController

# Conflicts:
#	main.py

j-siderius committed:
added places for zombie spawning and updating (check TODOs)

FrankBosman reviewed (approved):
Ziet er goed uit
